### PR TITLE
Call spokes service to generate id and support for an async at initializeCLI

### PR DESCRIPTION
**Pull request by jperata**

### CODE REVIEW

**Comment by jperata** [comment]
This is work in progress, still need to repair tests (and add the relevant tests), creating the PR for visibility and travis purposes

File: lib/client/bst-config.ts [hunk 1]
```diff
@@ -88,14 +88,36 @@ export class BSTConfig {
...
     }
 
-    private static createConfig(): any {
-        let nodeID = uuid.v4();
+    private static async createConfig(): Promise<any> {
         let lambdaConfig = LambdaConfig.defaultConfig().lambdaDeploy;
+        try {
+            const pipeInfo = await BSTConfig.createSpokesPipe();
+            return {
+                "endpoint": pipeInfo.endPoint.name,
```
Comment (jkelvie) [inline lib/client/bst-config.ts]: This should be labeled sourceID

File: lib/client/bst-config.ts [hunk 2]
```diff
@@ -19,8 +19,12 @@ export class BSTConfig {
      * Loads the configuration
      * Done all synchronously as this is done first thing at startup and everything waits on it
      */
-    public static load(): BSTConfig {
-        BSTConfig.bootstrapIfNeeded();
+    public static async load(): Promise<BSTConfig> {
+        try {
+            await BSTConfig.bootstrapIfNeeded();
+        } catch (error) {
+            throw error;
```
Comment (jperata) [inline lib/client/bst-config.ts]: According to a conversation with @jkelvie will put the logging information errors in a catch in the promise used on the bin files.

File: lib/client/bst-config.ts [hunk 3]
```diff
@@ -88,14 +88,36 @@ export class BSTConfig {
...
+    private static async createSpokesPipe(): Promise<any> {
+        let isUUIDUnassigned = false;
+        let sourceNameGenerator = null;
+        let spokesClient = null;
+        try {
+            sourceNameGenerator = new SourceNameGenerator();
+            const generatedKey = await sourceNameGenerator.callService();
+            spokesClient = new SpokesClient(generatedKey.id, generatedKey.secretKey);
+            isUUIDUnassigned = await spokesClient.verifyUUIDisNew();
+            if (isUUIDUnassigned) {
```
Comment (jperata) [inline lib/client/bst-config.ts]: Done

File: test/client/bst-config-test.ts [hunk 4]
```diff
@@ -66,12 +66,19 @@ describe("BSTConfig", function() {
 describe("BSTProcess", function() {
     describe("#run()", function() {
         let sandbox: SinonSandbox = null;
+
+        before(function () {
+            (<any> BSTProcess).processPath = function () {
```
Comment (chris-ramon) [inline test/client/bst-config-test.ts]: Perhaps `BSTProcess` should type asserted to [`class BSTProcess`](https://github.com/bespoken/bst/blob/67cee7462258807a8420e1d967912c479eca5ec7/lib/client/bst-config.ts#L102) ? — same in line [154](https://github.com/bespoken/bst/pull/235/commits/457a8727a5957845512c498848850a080b382529#diff-2ae95d89409189e7a5d7f6c3307fc8f4R154)

Comment (chris-ramon) [inline test/client/bst-config-test.ts]: we shouldn't, here we are monkey patching, all good :smile:

**Comment by chris-ramon** [comment]
all tests are passing! :tada:

**Pull request merged**